test/integration/protocol_integration_test.cc

```diff
@@ -1913,6 +1976,11 @@ name: encode-headers-return-stop-all-filter
                      downstreamProtocol() == Http::CodecClient::Type::HTTP1)
                         ? 1
                         : 2;
+  if (downstreamProtocol() == Http::CodecClient::Type::HTTP1 &&
+      upstreamProtocol() == FakeHttpConnection::Type::HTTP3) {
+    // TODO(alyssawilk) Figure out why the bytes mismatch with the test expectation below.
```

Comment:
you probably need to tweak times_added.
AFIK bytes added in trailers only come through for H2/H3 on both sides so times_added should be 2 if trailers work for both upstream and downstrema.

Reply:
The failed tests uses H1 for downstream, so I would expect times_added should be 1, but the test fails with:
[ RUN      ] UpstreamProtocols/DownstreamProtocolIntegrationTest.TestEncodeHeadersReturnsStopAllWatermark/IPv4_HttpDownstream_Http3Upstream
test/integration/protocol_integration_test.cc:1984: Failure
Expected equality of these values:
  count_ * size_ + added_decoded_data_size_ * times_added
    Which is: 70001
  response->body().size()
    Which is: 70002

So I suspect Http3 somehow added the twice.